dll/win32/cryptui/main.c

```diff
@@ -1280,13 +1284,21 @@ static LRESULT CALLBACK cert_mgr_dlg_proc(HWND hwnd, UINT msg, WPARAM wp,
         case IDCANCEL:
             free_certs(GetDlgItem(hwnd, IDC_MGR_CERTS));
             close_stores(GetDlgItem(hwnd, IDC_MGR_STORES));
+#ifndef __REACTOS__
             data = (struct CertMgrData *)GetWindowLongPtrW(hwnd, DWLP_USER);
             ImageList_Destroy(data->imageList);
             HeapFree(GetProcessHeap(), 0, data);
+#endif
             EndDialog(hwnd, IDCANCEL);
             break;
         }
         break;
+#ifdef __REACTOS__
+    case WM_DESTROY:
+        data = (struct CertMgrData *)GetWindowLongPtrW(hwnd, DWLP_USER);
+        HeapFree(GetProcessHeap(), 0, data);
```

Comment:
```suggestion
        ImageList_Destroy(data->imageList);
        HeapFree(GetProcessHeap(), 0, data);
```
It should be safe to do it there now.
Please move the WM_DESTROY handling just below the WM_INITDIALOG one for consistency.

Reply:
ImageList_Destroy(data->imageList);
remains unsafe as the listview will try to free it anyway.

OK to move WM_DESTROY